src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -778,15 +857,83 @@ protected int MakeSlot(BoundExpression node)
+                            if (fieldSymbol.IsStatic || receiverOpt == null || receiverOpt.Kind == BoundKind.TypeExpression) return null; // access of static field
+                            if (fieldSymbol.IsFixed) return null; // fixed buffers are not tracked
+                            if ((object)receiverOpt.Type == null || receiverOpt.Type.TypeKind != TypeKind.Struct) return null; // field of non-struct
+                            node = receiverOpt;
+                            break;
+                        }
+                    case BoundKind.EventAccess:
+                        {
+                            var eventAccess = (BoundEventAccess)node;
+                            var eventSymbol = eventAccess.EventSymbol;
+                            var receiverOpt = eventAccess.ReceiverOpt;
+                            if (eventSymbol.IsStatic || receiverOpt == null || receiverOpt.Kind == BoundKind.TypeExpression) return null; // access of static event
+                            if ((object)receiverOpt.Type == null || receiverOpt.Type.TypeKind != TypeKind.Struct) return null; // event of non-struct
+                            if (!eventSymbol.HasAssociatedField) return null;
+                            node = receiverOpt;
+                            break;
+                        }
+                    case BoundKind.PropertyAccess:
+                        {
+                            var propAccess = (BoundPropertyAccess)node;
+                            var propSymbol = propAccess.PropertySymbol;
+                            var receiverOpt = propAccess.ReceiverOpt;
+                            if (propSymbol.IsStatic || receiverOpt == null || receiverOpt.Kind == BoundKind.TypeExpression) return null; // access of static field
+                            if ((object)receiverOpt.Type == null || receiverOpt.Type.TypeKind != TypeKind.Struct) return null; // field of non-struct
+                            node = receiverOpt;
+                            break;
+                        }
+                    default:
+                        return null;
+                }
+            }
+        }
+
+        /// <summary>
         /// Check that the given variable is definitely assigned.  If not, produce an error.
         /// </summary>
-        /// <param name="symbol"></param>
-        /// <param name="node"></param>
         protected void CheckAssigned(Symbol symbol, CSharpSyntaxNode node)
         {
             Debug.Assert(!IsConditionalState);
             if ((object)symbol != null)
             {
+                NoteRead(symbol);
+
+                // If we're in a local func, instead of reporting an unassigned
+                // local, we should just note that we read it. We won't know if
+                // the local is actually unassigned until we get to the local
+                // function call
+                if (IsCapturedInLocalFunction(symbol))
+                {
+                    return;
```

Comment:
Oh, I see the previous statement notes the read.

---
In reply to: [73436691](https://github.com/dotnet/roslyn/pull/12888#discussion_r73436691) [](ancestors = 73436691)